app/models/ability.rb

```diff
@@ -0,0 +1,44 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, Post, author_id: user.id
+      can :create, Comment, author_id: user.id
+      can :create, Like
+      can :destroy, Post do |post|
```

Comment:
- [Optional] dear @smunoz1988, to simplify the matter and make your authorization rules a bit more clear, consider using `can :manage` to allow all action to a user. 
> Something like 
```ruby
can :manage, Post, author: user # This is a more efficient way of giving all the permission. 
```